code/edgeidentity/src/test/java/com/adobe/marketing/mobile/edge/identity/IdentityTests.java

```diff
@@ -269,6 +270,186 @@ public void call(Object o) {}
 		assertEquals(AdobeError.UNEXPECTED_ERROR, errorCapture.get(KEY_CAPTUREDERRORCALLBACK));
 	}
 
+	// ========================================================================================
+	// getUrlVariables API
+	// ========================================================================================
+	@Test
+	public void testGetUrlVariables() {
+		// setup
+		final ArgumentCaptor<Event> eventCaptor = ArgumentCaptor.forClass(Event.class);
+		final ArgumentCaptor<AdobeCallback> adobeCallbackCaptor = ArgumentCaptor.forClass(AdobeCallback.class);
+		final ArgumentCaptor<ExtensionErrorCallback> extensionErrorCallbackCaptor = ArgumentCaptor.forClass(
+			ExtensionErrorCallback.class
+		);
+		final List<String> callbackReturnValues = new ArrayList<>();
+
+		// test
+		Identity.getUrlVariables(
+			new AdobeCallback<String>() {
+				@Override
+				public void call(String s) {
+					callbackReturnValues.add(s);
```

Comment:
Do you need a latch here? This may work locally on a fast machine, but it may not complete in time on the slower CI systems.